src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -310,6 +310,145 @@ def k8s_install_cli(client_version='latest', install_location=None):
+                          location=None, service_principal=None, client_secret=None, url_chart=None,
+                          windows=False, linux=True):
+    """Deploy the ACI-Connector to a AKS cluster
+    :param name_aks: The name of the AKS cluster. The name is case insensitive.
+    :type name_aks: str
+    :param resource_group_name_aks: The name of the resource group where the AKS cluster is deployed.
+    :type resource_group_name_aks: str
+    :param name: The name for the ACI Connector
+    :type name: str
+    :param resource_group_name: The name of the resource group where to deploy the ACI instances.
+    :type resource_group_name: str
+    :param location:
+    :type location: str
+    :param service_principal: The service principal used for ACI authentication
+    to Azure APIs. If not specified, it is created for you and stored in the
+    ${HOME}/.azure directory.
+    :type service_principal: str
+    :param client_secret: The secret associated with the service principal. If
+    --service-principal is specified, then secret should also be specified. If
+    --service-principal is not specified, the secret is auto-generated for you
+    and stored in ${HOME}/.azure/ directory.
+    :type client_secret: str
+    :param url_chart: URL to the chart,
+    Default : https://github.com/Azure/aci-connector-k8s/raw/master/charts/aci-connector.tgz
+    :type url_chart: str
+    :param windows: Os type target will be Windows for the connector
+    :type windows: bool
+    :param linux: Os type target will be Linux for the connector
+    :type linux: bool
+    """
+    from subprocess import PIPE, Popen
+    helm_not_installed = "Error : Helm not detected, please verify if it is installed."
+    image_tag = 'latest'
+    if url_chart is None:
+        chart_url = "https://github.com/Azure/aci-connector-k8s/raw/master/charts/aci-connector.tgz"
+    else:
+        chart_url = url_chart
+    # Check if Helm is installed locally
+    try:
+        p = Popen(["helm"], stdout=PIPE, stderr=PIPE)
+    except OSError:
+        raise CLIError(helm_not_installed)
+    # Get the credentials from a AKS instance
+    _, browse_path = tempfile.mkstemp()
+    aks_get_credentials(client, resource_group_name_aks, name_aks, admin=False, path=browse_path)
+    subscription_id = _get_subscription_id()
+    dns_name_prefix = _get_default_dns_prefix(name, resource_group_name, subscription_id)
+    # Validate if the RG exists
+    groups = _resource_client_factory().resource_groups
+    # Just do the get, we don't need the result, it will error out if the group doesn't exist.
+    rgkaci = groups.get(resource_group_name)
+    # Auto assign the location
+    if location is None:
+        location = rgkaci.location  # pylint:disable=no-member
+    # Ensure that the SPN exists
+    principal_obj = _ensure_service_principal(service_principal, client_secret, subscription_id,
```

Comment:
I have a concern about it: `_ensure_service_principal` implicitly grants subscription scope contributor to the spn. I dont think you want it.
Suggest to remove [below code](https://github.com/julienstroheker/azure-cli/blob/ba59b596ce4e1ff1d00276f37bf8c59145eea00b/src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py#L1500) from `_ensure_service_principal` all together as it's not needed:
```python
# add role first before save it
if not _add_role_assignment('Contributor', service_principal):
     logger.warning('Could not create a service principal with the right permissions. '
                 'Are you an Owner on this project?')
```

Reply:
@mboersma

Reply:
@weinong You are asking @julienstroheker to remove code thats not even part of the commit. If theres a concern with another module/function i suggest submitting a PR or issue for that code. This is also an overall problem that should be addressed as an issue regarding the functionality of that method. Which I agree, we should not be scoping to an entire sub, but we are already doing that for acs/aks, thus this should be its own issue and un-related to thi pr.

Reply:
@listonb np! we can fix that separately.